sources/en/c/chrysanthemumgarden.py

```diff
@@ -44,8 +62,24 @@ def read_novel_info(self):
 
         self.volumes = [{"id": x, "title": ""} for x in volumes]
 
+    def login(self, email, password):
+        self.password = password
+
     def download_chapter_body(self, chapter):
-        soup = self.get_soup(chapter["url"])
+
+        chapter_url = chapter["url"]
+        soup = self.get_soup(chapter_url)
+
+        if soup.select_one("#site-pass"):
+            payload = {
+                "site-pass": self.password,
+                "nonce-site-pass": soup.select_one("#nonce-site-pass")["value"],
+                "_wp_http_referer": urlparse(chapter_url).path,
+            }
+
+            soup = self.make_soup(
+                self.submit_form(url=self.absolute_url(chapter_url), data=payload),
```

Comment:
```suggestion
                self.submit_form(url=self.absolute_url(chapter_url), data=payload, multipart=True),
```

Reply:
you can also pass `headers` and set stuff like `referer` and `user-agent` there in case they do check those

Reply:
I don't think they check theses here as it works as is but that's good to know